File: app/src/test/java/ml/melun/mangaview/model/UrlUpdateResultTest.java
package ml.melun.mangaview.model;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class UrlUpdateResultTest {
    @Test
    public void exposesSuccessAndUrlForJavaObservers() {
        UrlUpdateResult result = new UrlUpdateResult(
                true,
                "https://example.test/cm",
                "https://request.example.test");

        assertTrue(result.getSuccess());
        assertEquals("https://example.test/cm", result.getUrl());
        assertEquals("https://request.example.test", result.getRequestUrl());
        assertTrue(result.isForRequest("https://request.example.test"));
        assertFalse(result.isForRequest("https://stale.example.test"));
        assertFalse(result.isForRequest(null));
    }
}

File: app/src/test/java/ml/melun/mangaview/model/UrlUpdateResultTest.kt
package ml.melun.mangaview.model

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test

class UrlUpdateResultTest {
    @Test
    fun exposesSuccessAndUrlForJavaObservers() {
        val result = UrlUpdateResult(true, "https://example.test/cm")

        assertTrue(result.success)
        assertEquals("https://example.test/cm", result.url)
    }
}
